cli/src/program.rs

```diff
@@ -2740,6 +2769,44 @@ fn report_ephemeral_mnemonic(words: usize, mnemonic: bip39::Mnemonic) {
     eprintln!("[BUFFER_ACCOUNT_ADDRESS] argument to `solana program close`.\n{divider}");
 }
 
+fn set_compute_budget_ixs_if_needed(ixs: &mut Vec<Instruction>, compute_unit_price: Option<u64>) {
+    let Some(compute_unit_price) = compute_unit_price else {
+        return;
+    };
+
+    let mut compute_unit_limit: u32 = 300; // 2 * 150 for compute budget ixs
+    for ix in ixs.iter() {
+        if ix.program_id == bpf_loader_upgradeable::id() {
+            compute_unit_limit += 2370;
+            match ix.data.first() {
+                Some(2) // DeployWithMaxDataLen
+                | Some(6) // ExtendProgram
+                => {
+                    // add compute for native system program invocation
+                    compute_unit_limit += 150;
+                }
+                _ => {}
+            }
+        } else if ix.program_id == system_program::id() {
+            compute_unit_limit += 150;
+        } else {
+            panic!(
+                "Couldn't estimate compute unit limit for unexpected program {}",
+                ix.program_id
+            );
+        }
+    }
```

Comment:
I'm not really in favor of having compute unit limit logic here in the cli client. I'd prefer to get the units consumed from the simulate transaction call.

Reply:
I would feel better knowing that deploys won't get broken in the future if we increase the compute unit limit for certain instructions. Let me know what you think of the simulation code!